crates/fxp/src/dynamic.rs

```diff
@@ -331,6 +494,35 @@ impl Ord for DynamicDecimal {
     }
 }
 
+/// Supports uniform sampling for DynamicDecimals with a specified scale.
+/// This is almost like UniformSampler, but we need to pass additional information
+/// to the constructor, so we cannot use that trait.
+#[derive(Clone, Debug)]
+pub struct UniformDecimal {
+    significand: UniformInt<i128>,
+    scale: u8,
+}
+
+impl UniformDecimal {
+    /// Create a new UniformDecimal
+    pub fn new(low: DynamicDecimal, high: DynamicDecimal, scale: u8) -> Self {
+        if low >= high {
+            panic!("Invalid range");
+        }
+        let pow = pow10(scale as usize);
+        Self {
+            significand: UniformInt::new(&low.significand() * pow, &high.significand() * pow),
+            scale,
+        }
```

Comment:
I don't understand this at all. If `low` and `high` have different exponents, then it doesn't make sense to choose a value between their significands. And I don't understand why this is multiplying by `10**scale`.

What are the desired semantics of this function?

Reply:
This uses the uniform generation for i128.
Initially I wrote `assert!(low.exponent() == high.exponent())`, and used that for 'scale', but that does not hold, because `DynamicDecimal` actually normalizes the exponent to have a canonical representation.
I want uniform numbers between -10.00 and 10.00, with 2 decimals. So I generate integers uniformly between -1000 and 1000, and scale down them down by 2 digits.

Reply:
Suppose I pass in low = 2.5, high = 5, scale = 2. low will have significand 25, exponent 1. high will have significand 5, exponent 0. This will call UniformInt::new(2500, 500), which doesn't make sense.